packages/react-components/src/components/trace-context-component.tsx

```diff
@@ -193,13 +198,61 @@ export class TraceContextComponent extends React.Component<TraceContextProps, Tr
         this.props.messageManager.removeStatusMessage(this.INDEXING_STATUS_BAR_KEY);
         this.props.messageManager.removeStatusMessage(this.TIME_SELECTION_STATUS_BAR_KEY);
         this.props.removeResizeHandler(this.onResize);
+        signalManager().off(Signals.TIMEGRAPH_ZOOMED, this.onTimeGraphZoomed);
+        signalManager().off(Signals.TIMEGRAPH_RESET, this.onTimeGraphReset);
     }
 
     async componentDidUpdate(): Promise<void> {
         // Rebuild enables tooltip on newly added output component
         ReactTooltip.rebuild();
     }
 
+    private doHandleTimeGraphZoomedSignal(hasZoomedIn: boolean) {
+        if (hasZoomedIn) {
+            this.zoomButton(hasZoomedIn);
+        } else {
+            this.zoomButton(hasZoomedIn);
+        }
+    }
+
+    private doHandleTimeGraphResetSignal() {
+        this.unitController.viewRange = { start: 0, end: this.unitController.absoluteRange };
+    }
+
+    private zoomButton(zoomIn: boolean) {
+        let newStartRange = 0;
+        let newEndRange = 0;
+        let position = (this.unitController.viewRange.end + this.unitController.viewRange.start) / 2;
+        const percentZoom = 0.9;
+        if (this.unitController.selectionRange) {
+            if (this.unitController.selectionRange.end < this.unitController.selectionRange.start){
+                const temp = this.unitController.selectionRange.end;
+                this.unitController.selectionRange.end = this.unitController.selectionRange.start;
+                this.unitController.selectionRange.start = temp;
```

Comment:
You should not change the user's selectionRange. You could use local variables, but really you don't need to ensure they are start < end, just adjust the code below to handle this case.